src/services/refactors/moveToFile.ts

```diff
@@ -0,0 +1,1114 @@
+    const body = addExports(oldFile, toMove.all, usage.oldFileImportsFromNewFile, useEsModuleSyntax);
+    if (newFileExists) {
+        const sourceFile = program.getSourceFile(newFile);
+        if (sourceFile && sourceFile.statements.length > 0) {
+            changes.insertNodesAfter(sourceFile, sourceFile.statements[sourceFile.statements.length - 1], body);
+        }
+        if (sourceFile && imports.length > 0) {
+            insertImports(changes, sourceFile, imports, /*blankLineBetween*/ true, preferences);
+        }
+    }
+    if (importAdder) {
+        importAdder.writeFixes(changes);
+    }
+    if (imports.length && body.length) {
+        return [
+            ...prologueDirectives,
+            ...imports,
+            SyntaxKind.NewLineTrivia as const,
+            ...body
+        ];
+    }
+
+    return [
+        ...prologueDirectives,
+        ...imports,
+        ...body,
+    ];
+}
+
+function getNewFileImportsAndAddExportInOldFile(
+    oldFile: SourceFile,
+    newFile: string,
+    importsToCopy: Map<Symbol, boolean>,
+    newFileImportsFromOldFile: Set<Symbol>,
+    changes: textChanges.ChangeTracker,
+    checker: TypeChecker,
+    program: Program,
+    host: LanguageServiceHost,
+    useEsModuleSyntax: boolean,
+    quotePreference: QuotePreference,
+    importAdder?: codefix.ImportAdder,
+): readonly AnyImportOrRequireStatement[] {
+    const copiedOldImports: AnyImportOrRequireStatement[] = [];
+    if (importAdder) {
+        importsToCopy.forEach((isValidTypeOnlyUseSite, symbol) => {
+            try {
+                importAdder.addImportFromExportedSymbol(skipAlias(symbol, checker), isValidTypeOnlyUseSite);
+            }
+            catch {
+                for (const oldStatement of oldFile.statements) {
+                    forEachImportInStatement(oldStatement, i => {
+                        append(copiedOldImports, filterImport(i, factory.createStringLiteral(moduleSpecifierFromImport(i).text), name => importsToCopy.has(checker.getSymbolAtLocation(name)!)));
+                    });
+                }
+            }
+        });
+    }
+    else {
+        for (const oldStatement of oldFile.statements) {
+            forEachImportInStatement(oldStatement, i => {
+                const moduleSpecifier = moduleSpecifierFromImport(i);
+                const resolved = oldFile.resolvedModules?.get(moduleSpecifier.text, getModeForUsageLocation(oldFile, moduleSpecifier));
+                const fileName = resolved?.resolvedModule?.resolvedFileName;
+                if (fileName) {
+                    const sourceFile = program.getSourceFile(newFile);
+                    if (sourceFile) {
+                        const pathToNewFileWithExtension = resolvePath(getDirectoryPath(oldFile.path), fileName);

```

Comment:
I’m pretty sure `pathToNewFileWithExtension === fileName` 100% of the time. Are you sure this is necessary? `getModuleSpecifier` just wants the file name of the file you’re importing, which you already have with `fileName`.